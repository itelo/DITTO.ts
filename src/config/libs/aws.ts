import aws from "aws-sdk";
import configStack from "@config/index";

let instance: any;
class Aws {
  constructor() {
    if (instance) {
      return instance;
    }

    instance = this;
    return this;
  }

  public s3() {
    const config = configStack.config;
    aws.config.update({
      accessKeyId: config.aws.s3.accessKeyId,
      secretAccessKey: config.aws.s3.secretAccessKey
    });
    return new aws.S3();
  }
}

export default new Aws();
