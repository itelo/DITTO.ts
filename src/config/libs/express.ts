import { ConfigEnvsObject } from "types/config/env";
import express, { Application, Request, Response, NextFunction } from "express";
import morgan from "morgan";
import cons from "consolidate";
import bodyParser from "body-parser";
// import mongoose from "mongoose";
// import MongoStore from "connect-mongo";
import compress from "compression";
import methodOverride from "method-override";
import cookieParser from "cookie-parser";
import helmet from "helmet";
// import flash from "connect-flash";
import path from "path";
import lodash from "lodash";
import lusca from "lusca";
import i18next, { Resource } from "i18next";
import cors from "cors";
import logger from "./logger";
import * as responses from "@utils/formaters/responses";
import { Codes } from "@utils/constants/codes";
import { HttpStatus } from "@utils/constants/httpStatus";

/**
 * Initialize local variables
 */
function initLocalVariables(app: Application, config: ConfigEnvsObject) {
  // Setting application local variables
  app.locals.title = config.app.title;
  app.locals.description = config.app.description;
  if (config.secure && config.secure.ssl === true) {
    app.locals.secure = config.secure.ssl;
  }
  app.locals.keywords = config.app.keywords;
  app.locals.googleAnalyticsTrackingID = config.app.googleAnalyticsTrackingID;
  // app.locals.facebookAppId = config.facebook.clientID;
  // app.locals.twitterUsername = config.twitter.username;
  // app.locals.jsFiles = config.files.client.js;
  // app.locals.cssFiles = config.files.client.css;
  // app.locals.livereload = config.livereload;
  // app.locals.logo = config.logo;
  // app.locals.favicon = config.favicon;
  app.locals.env = process.env.NODE_ENV;
  app.locals.domain = config.domain;

  // Passing the request url to environment locals
  app.use(function(req: Request, res: Response, next: NextFunction) {
    res.locals.host = req.protocol + "://" + req.hostname;
    res.locals.url = req.protocol + "://" + req.headers.host + req.originalUrl;
    next();
  });
}

/**
 * Initialize application middleware
 */
function initMiddleware(app: Application, config: ConfigEnvsObject) {
  //
  app.use(
    cors({
      origin: "http://localhost:3001"
    })
  );

  // Should be placed before express.static
  app.use(
    compress({
      filter(req: Request, res: Response) {
        // @ts-ignore
        return /json|text|javascript|css|font|svg/.test(
          res.get("Content-Type")
        );
      },
      level: 9
    })
  );

  // Enable logger (morgan) if enabled in the configuration file
  if (lodash.has(config, "log.format")) {
    app.use(morgan(logger.getLogFormat(), logger.getMorganOptions()));
  }

  // Environment dependent middleware
  if (process.env.NODE_ENV === "development") {
    // Disable views cache
    app.set("view cache", false);
  } else if (process.env.NODE_ENV === "production") {
    app.locals.cache = "memory";
  }

  // Request body parsing middleware should be above methodOverride
  app.use(
    bodyParser.urlencoded({
      extended: true
    })
  );
  app.use(bodyParser.json());
  app.use((err: any, req: Request, res: Response, next: NextFunction) => {
    if (
      err.constructor === SyntaxError &&
      err.status >= 400 &&
      err.status < 500 &&
      err.message.indexOf("JSON")
    ) {
      console.error(err);

      responses.sendError(
        res,
        Codes.REQUEST__MISSING_PARAMS,
        "Malformed JSON input",
        HttpStatus.UNPROCESSABLE_ENTITY
      );
    }
  });
  app.use(methodOverride());

  // Add the cookie parser and flash middleware
  app.use(cookieParser());
  // app.use(flash());
}

/**
 * Invoke modules server configuration
 */
function initModulesConfiguration(app: Application, config: ConfigEnvsObject) {
  config.files.configs.forEach(async (configPath: string) => {
    const modules = await import(path.resolve(configPath));
    modules.default(app);
  });
}

/**
 * Configure Helmet headers configuration for security
 */
function initHelmetHeaders(app: Application) {
  // six months expiration period specified in seconds
  const SIX_MONTHS = 15778476;

  app.use(helmet.frameguard());
  app.use(helmet.xssFilter());
  app.use(helmet.noSniff());
  app.use(helmet.ieNoOpen());
  app.use(
    helmet.hsts({
      maxAge: SIX_MONTHS,
      includeSubdomains: true,
      force: true
    })
  );
  app.disable("x-powered-by");
}

/**
 * Configure the modules ACL policies
 */
function initModulesServerPolicies(config: ConfigEnvsObject) {
  // Globbing policy files
  config.files.policies.forEach(async (policyPath: string) => {
    const policies = await import(path.resolve(policyPath));

    policies.invokeRolesPolicies();
  });
}

// /**
//  * Configure the modules validations
//  */
// function initModulesServerValidations(config: ConfigEnvsObject) {
//   // Globbing policy files
//   config.files.validations.forEach(async (validationPath: string) => {
//     const validations = await import(path.resolve(validationPath));
//     // policies.invokeRolesPolicies();
//   });
// }

/**
 * Configure the modules server routes
 */
function initModulesServerRoutes(app: Application, config: ConfigEnvsObject) {
  // Globbing routing files
  config.files.routes.forEach(async (routePath: string) => {
    const routes = await import(path.resolve(routePath));

    routes.default(app);
  });
}

/**
 * Configure view engine
 */
function initViewEngine(app: Application) {
  // assign the swig engine to .html files
  app.engine("html", cons.swig);

  // set .html as the default extension
  app.set("view engine", "html");
  app.set("views", path.resolve("./"));
}

/**
 * Configure the static routes
 */
function initStaticRoutes(app: Application) {
  // Setting the app router and static folder
  app.use(
    "/public",
    express.static(path.resolve("./public"), { maxAge: 86400000 })
  );
}

/**
 * Configure Graphql
 */
function configureGraphql(app: Application, config: ConfigEnvsObject) {
  // Load the Socket.io configuration
  const server = require("./graphqlv2").default(app, config);

  // Return server object
  return server;
}

/**
 * Configure Socket.io
 */
function configureSocketIO(app: Application) {
  // Load the Socket.io configuration
  const server = require("./socket.io").default(app);

  // Return server object
  return server;
}

/**
 * Configure i18n
 */
async function initI18N(app: Application, config: ConfigEnvsObject) {
  app.use(function(req, res, next) {
    const lng = req.get("content-language");
    if (lng) i18next.changeLanguage(lng);

    next();
  });

  const resources = {} as Resource;

  for (const langPath of config.files.i18n) {
    const helper = langPath.split("/");
    const langName = helper[helper.length - 1].split(".").shift();
    if (langName) {
      const translation = await import(path.resolve(langPath));
      resources[langName] = { translation: translation.default };
    }
  }

  i18next.init(
    {
      fallbackLng: "pt-BR",
      resources,
      debug: false
    },
    (err: Error) => {
      if (err) {
        throw `something went wrong loading ${err}`;
      }
    }
  );
}

/**
 * Initialize the Express application
 */
export function init(config: ConfigEnvsObject) {
  // Initialize express app
  const app = express();

  // Initialize local variables
  initLocalVariables(app, config);

  // Initialize Express middleware
  initMiddleware(app, config);

  // Initialize Helmet security headers
  initHelmetHeaders(app);

  // Initialize Modules configuration
  initModulesConfiguration(app, config);

  // Initialize modules server authorization policies
  initModulesServerPolicies(config);

  // Initialize modules server routes
  initModulesServerRoutes(app, config);

  // Initialize View Engine
  initViewEngine(app);

  initStaticRoutes(app);

  // init
  initI18N(app, config);

  // configure graphql
  const server = configureGraphql(app, config);

  return server;
}
