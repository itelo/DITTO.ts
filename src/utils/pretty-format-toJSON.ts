// my-serializer-module
const prettyFormatToJSON = {
  print(val: any, serialize: (val: any) => string, indent: any) {
    return serialize(JSON.stringify(val, undefined, 2));
  },

  test(value: any) {
    console.log({ value });

    return value && value.input && value.output;
  }
};

export { prettyFormatToJSON };
