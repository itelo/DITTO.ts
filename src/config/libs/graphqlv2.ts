import { Application, Request, Response } from "express";
import { ConfigEnvsObject } from "types/config/env";
import { makeExecutableSchema } from "graphql-tools";
import { createServer } from "http";

import { mergeTypes } from "merge-graphql-schemas";
// import { graphqlExpress, graphiqlExpress } from "graphql-server-express";
import { ApolloServer } from "apollo-server-express";
import { printSchema } from "graphql/utilities/schemaPrinter";
import { importSchema } from "graphql-import";
import lodash from "lodash";
import { DocumentNode } from "graphql";
// ensure type
import "passport";

/**
 * Configure Graphql
 */
function configureGraphql(app: Application, config: ConfigEnvsObject) {
  console.log({ typedefFilePath: config.files.typedefs });
  // console.log({ typedefFilePath: config.files.typedefs });

  const typeDefs = config.files.typedefs.map<DocumentNode>(
    (typedefFilePath: string) =>
      require(typedefFilePath.replace("src/", "@")).default
  );

  console.log({ typeDefs });

  const resolversFiles = config.files.resolvers.map<string>(
    (resolverFilePath: string) =>
      require(resolverFilePath.replace("src/", "@")).default
  );

  // const typeDefs = Promise.all(typedefsFilesPromise);
  // const resolversFiles = Promise.all(resolversFilesPromise);
  // const resolversFiles = resolversFilesWithDefault.map(file => file.default);

  const resolvers = lodash.merge(resolversFiles);
  // const typeDefs = typedefsFilesWithDefault.map(file => file.default);

  // GraphQL: Schema
  const server = new ApolloServer({
    typeDefs,
    resolvers,
    context: ({ req }) => {
      // console.log(req);
      if (req) {
        // add the user to the context
        return { req, user: req.user };
      }

      return {};
    },
    subscriptions: {
      onConnect: (connectionParams, webSocket) => {
        // console.log({ connectionParams, webSocket });
        // if (connectionParams.authToken) {
        //   return validateToken(connectionParams.authToken)
        //     .then(findUser(connectionParams.authToken))
        //     .then(user => {
        //       return {
        //         currentUser: user,
        //       };
        //     });
        // }
        // throw new Error('Missing auth token!');
      }
    },
    // http://${config.host}:${config.port}
    playground: {
      endpoint: `/graphiql`,
      settings: {
        "general.betaUpdates": false,
        "editor.cursorShape": "line",
        "editor.fontSize": 14,
        "editor.fontFamily":
          "'Source Code Pro', 'Consolas', 'Inconsolata', 'Droid Sans Mono', 'Monaco', monospace",
        "editor.theme": "dark",
        "editor.reuseHeaders": true,
        "prettier.printWidth": 80,
        "request.credentials": "omit",
        "tracing.hideTracingResponse": true
      }
    }
  });

  server.applyMiddleware({ app, path: "/graphiql" });

  const httpServer = createServer(app);

  server.installSubscriptionHandlers(httpServer);

  return httpServer;
  // const schema = makeExecutableSchema({
  //   typeDefs,
  //   resolvers
  // });

  // app.use(
  //   "/api/graphql",
  //   graphqlExpress((req: Request, res: Response) => ({ schema }))
  // );
  // // context: { req, res, user: req.user },

  // // /api/graphiql
  // app.use(
  //   "/graphiql",
  //   graphiqlExpress({
  //     endpointURL: "/api/graphql"
  //     // subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`
  //   })
  // );

  // // /api/schema
  // app.use("/schema", (_: Request, res: Response) => {
  //   res.set("Content-Type", "text/plain");
  //   res.send(printSchema(schema));
  // });
}

export default configureGraphql;
