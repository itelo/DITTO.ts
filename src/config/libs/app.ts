import { Application } from "express";
import { ConfigEnvsObject } from "types/config/env";
import chalk from "chalk";
import configStack from "@config/index";
import * as mongooseService from "@config/libs/mongoose";
import * as express from "@config/libs/express";

export function init(config: ConfigEnvsObject, callback: Function) {
  // if (process.env.NODE_ENV === "test") {
  // mongooseService.connectLoadAndSeed(config, (db: any) => {
  //   const app = express.init(config);
  //   if (callback) callback(app, db, config);
  // });
  // } else {
  mongooseService.connect(config, (db: any) => {
    // Initialize Models
    mongooseService.loadModels(config);

    // Initialize express
    const app = express.init(config);
    if (callback) callback(app, db, config);
  });
  // }
}

export function start(callback?: Function) {
  console.log("NODE_ENV => ", process.env.NODE_ENV);
  const config = configStack.config;

  // console.log(config);

  init(config, (app: Application, db: any, config: any) => {
    // Start the app by listening on <port> at <host>
    app.listen(config.port, config.host, () => {
      // Create server URL
      const server = `${(process.env.NODE_ENV === "secure"
        ? "https://"
        : "http://") + config.host}:${config.port}`;
      // Logging initialization
      console.log("--");
      console.log(chalk.green(config.app.title));
      console.log();
      console.log(chalk.green(`Environment:     ${process.env.NODE_ENV}`));
      console.log(chalk.green(`Server:          ${server}`));
      process.env.NODE_ENV !== "production" &&
        console.log(chalk.green(`Database:        ${config.db.uri}`));
      console.log(chalk.green(`App version:     ${config.pkg.version}`));
      if (config.pkg["pkg-version"])
        console.log(
          chalk.green(`DITTO.JS version: ${config.pkg["pkg-version"]}`)
        );
      console.log("--");

      if (callback) callback(app, db, config);
    });
  });
}

export function cleanStart(): Promise<Application> {
  const config = configStack.config;

  return new Promise(resolve => {
    return init(config, (app: Application) => resolve(app));
  });
}
