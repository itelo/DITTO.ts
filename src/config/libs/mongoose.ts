import lodash from "lodash";
import chalk from "chalk";
import path from "path";
import mongoose from "mongoose";
import MongoMemoryServer from "mongodb-memory-server";

import * as seed from "@config/libs/mongo-seed";
import { ConfigEnvsObject } from "types/config/env";
// Load the mongoose models

function getConnectionString() {
  return new Promise(async (resolve, reject) => {
    const mongoServer = new MongoMemoryServer();
    try {
      const mongoURI = await mongoServer.getConnectionString();
      const memoryOptions = {
        autoReconnect: true,
        reconnectTries: Number.MAX_VALUE,
        reconnectInterval: 1000
      };
      resolve({ mongoURI, memoryOptions });
    } catch (err) {
      console.log(err);
      reject(err);
    }
  });
}

export function seedDB(config: ConfigEnvsObject) {
  // return seed.start();
  // @ts-ignore
  if (config.seedDB && config.seedDB.seed) {
    // console.log(chalk.bold.red("Warning:  Database seeding is turned on"));
    return seed.start();
  }
  // console.log(
  // chalk.bold.yellowBright("Warning:  Database seeding is turned off")
  // );
}
export function loadModels(config: ConfigEnvsObject, callback?: Function) {
  // Globbing model files
  config.files.models.forEach((modelPath: string) => {
    require(path.resolve(modelPath));
  });

  if (callback) callback();
}
// Initialize Mongoose
export async function connect(config: ConfigEnvsObject, callback?: Function) {
  mongoose.Promise = config.db.promise;

  let mongoURI = config.db.uri;
  let memoryOptions = {};

  // @ts-ignore
  if (config.db.inMemory) {
    const result = await getConnectionString();
    // @ts-ignore
    mongoURI = result.mongoURI;
    // @ts-ignore
    memoryOptions = result.memoryOptions;
  }

  console.log({ mongoURI });

  const options: any = lodash.merge(
    config.db.options || {},
    {
      useNewUrlParser: true
    },
    memoryOptions
  );

  mongoose
    .connect(mongoURI, options)
    // @ts-ignore
    .then(() => {
      // Enabling mongoose debug mode if required
      mongoose.set("debug", config.db.debug);
      // Call callback FN
      if (callback) callback(mongoose.connection.db);
    })
    .catch((err?: Error) => {
      console.log(err);
      console.error(chalk.red("Could not connect to MongoDB!"));
      if (callback) {
        if (err) {
          callback(undefined, err);
        } else {
          callback(undefined, "Could not connect to MongoDB!");
        }
      }
    });
}

export async function connectLoadAndSeed(
  config: ConfigEnvsObject,
  cb: Function
) {
  // return new Promise()
  // console.info(chalk.blueBright("Connecting to DB"));
  connect(
    config,
    (db: any) =>
      loadModels(config, () => {
        seed
          .start()
          .then(() => {
            cb(db);
          })
          .catch(err => {
            console.log(chalk.bgGreenBright(chalk.red(err)));
          });
      })
  );
}

export function disconnect(callback?: Function) {
  mongoose.connection.close((err: Error) => {
    if (process.env.NODE_ENV !== "test") {
      console.info(chalk.yellow("Disconnected from MongoDB."));
    }
    if (callback) return callback(err);
  });
}
