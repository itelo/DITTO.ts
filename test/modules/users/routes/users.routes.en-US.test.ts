import { Application } from "express";
import request from "supertest";
import configStack from "@config/index";
import * as server from "@config/libs/app";
import * as db from "@config/libs/mongoose";
import { prettyFormatToJSON } from "@utils/pretty-format-toJSON";
expect.addSnapshotSerializer(prettyFormatToJSON);

jest.setTimeout(10000);

let app: Application;

describe("route:/api/v1/signup", () => {
  let token: string;
  beforeAll(async done => {
    app = await server.cleanStart();
    done();
  });

  afterAll(done => {
    db.disconnect(done);
  });

  it("success: all data correct", done => {
    const doc = {
      input: {
        body: {
          first_name: "ilu2",
          last_name: "minate2",
          phone: "1234567891",
          email: "iluminate@india2.com",
          password: "!(_Um1n4t3"
        }
      },
      output: {}
    };
    request(app)
      .post("/api/v1/signup")
      .send(doc.input.body)
      .expect(200)
      .end((err, res) => {
        const output = {
          body: res.body
        };

        expect(res.body.data.user).toHaveProperty("first_name", "ilu2");
        expect(res.body.data.user).toHaveProperty("last_name", "minate2");
        expect(res.body.data.user).toHaveProperty(
          "email",
          "iluminate@india2.com"
        );
        expect(res.body.data.user).toHaveProperty("provider", "local");
        expect(res.body.data.user).toHaveProperty("roles", ["user"]);
        expect(res.body.data.user).toHaveProperty("_id");
        expect(res.body.data.user).toHaveProperty("profile_image_urls");
        expect(res.body.data).toHaveProperty("token");

        doc.output = output;

        expect(doc).toMatchSnapshot();

        if (err) return done(err);
        done();
      });
  });

  it("fail: due to blank first name", done => {
    const doc = {
      input: {
        body: {
          first_name: "",
          last_name: "minate",
          phone: "1234567891",
          email: "aaailuminate@india2.com",
          password: "!(_Um1n4t3"
        }
      },
      output: {}
    };
    request(app)
      .post("/api/v1/signup")
      .send(doc.input.body)
      .expect(422)
      .set("content-Language", "en-US")
      .end((err, res) => {
        const output = {
          body: res.body
        };

        doc.output = output;
        expect(doc).toMatchSnapshot();
        expect(res.body.error.message).toBe("First name is required field");

        if (err) return done(err);
        done();
      });
  });

  it("blank last name /api/v1/signup", done => {
    request(app)
      .post("/api/v1/signup")
      .send({
        first_name: "ilu",
        last_name: "",
        phone: "1234567891",
        email: "aaailuminate@india2.com",
        password: "!(_Um1n4t3"
      })
      .expect(422)
      .set("content-Language", "en-US")
      .end((err, res) => {
        // expect(res.body).toMatchSnapshot();
        expect(res.body.error.message).toBe("Last name is required field");

        if (err) return done(err);
        done();
      });
  });

  it("blank email /api/v1/signup", done => {
    request(app)
      .post("/api/v1/signup")
      .send({
        first_name: "ilu",
        last_name: "minate",
        phone: "1234567891",
        email: "",
        password: "!(_Um1n4t3"
      })
      .expect(422)
      .set("content-Language", "en-US")
      .end((err, res) => {
        // expect(res.body).toMatchSnapshot();
        expect(res.body.error.message).toBe(
          "The email you passed is not a valid one"
        );

        if (err) return done(err);
        done();
      });
  });

  it("test /api/v1/signin", done => {
    request(app)
      .post("/api/v1/signin")
      .send({
        email: "iluminate@india2.com",
        password: "!(_Um1n4t3"
      })
      .expect(200)
      .end((err, res) => {
        // expect(res.body).toMatchSnapshot();
        token = `${configStack.config.jwt.prefix} ${res.body}`;

        if (err) return done(err);
        done();
      });
  });

  it("test /api/v1/users/me", done => {
    request(app)
      .get("/api/v1/users/me")
      .expect(200)
      .set({ Authorization: token })
      .end((err, res) => {
        // expect(res.body).toMatchSnapshot();
        if (err) return done(err);
        done();
      });
  });
});
