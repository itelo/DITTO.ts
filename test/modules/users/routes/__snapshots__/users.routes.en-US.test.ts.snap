// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`route:/api/v1/signup fail: due to blank first name 1`] = `
"{
  \\"input\\": {
    \\"body\\": {
      \\"first_name\\": \\"\\",
      \\"last_name\\": \\"minate\\",
      \\"phone\\": \\"1234567891\\",
      \\"email\\": \\"aaailuminate@india2.com\\",
      \\"password\\": \\"!(_Um1n4t3\\"
    }
  },
  \\"output\\": {
    \\"body\\": {
      \\"success\\": false,
      \\"error\\": {
        \\"message\\": \\"Phone alreadyExists\\",
        \\"status\\": 422,
        \\"code\\": \\"auth/unique-already-in-use\\"
      }
    }
  }
}"
`;

exports[`route:/api/v1/signup success: all data correct 1`] = `
"{
  \\"input\\": {
    \\"body\\": {
      \\"first_name\\": \\"ilu2\\",
      \\"last_name\\": \\"minate2\\",
      \\"phone\\": \\"1234567891\\",
      \\"email\\": \\"iluminate@india2.com\\",
      \\"password\\": \\"!(_Um1n4t3\\"
    }
  },
  \\"output\\": {
    \\"body\\": {
      \\"success\\": true,
      \\"data\\": {
        \\"user\\": {
          \\"profile_image_urls\\": {
            \\"original\\": \\"/public/images/common/default.png\\",
            \\"x256\\": \\"/public/images/common/default.png\\",
            \\"x720\\": \\"/public/images/common/default.png\\"
          },
          \\"first_name\\": \\"ilu2\\",
          \\"last_name\\": \\"minate2\\",
          \\"email\\": \\"iluminate@india2.com\\",
          \\"roles\\": [
            \\"user\\"
          ],
          \\"_id\\": \\"5cb28df739a45ecc834c5cec\\",
          \\"phone\\": \\"1234567891\\",
          \\"created_at\\": \\"2019-04-14T01:33:43.769Z\\",
          \\"albums\\": [],
          \\"provider\\": \\"local\\",
          \\"__v\\": 0
        },
        \\"token\\": \\"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImlsdW1pbmF0ZUBpbmRpYTIuY29tIiwicHJvZmlsZV9pbWFnZV91cmxzIjp7Im9yaWdpbmFsIjoiL3B1YmxpYy9pbWFnZXMvY29tbW9uL2RlZmF1bHQucG5nIiwieDI1NiI6Ii9wdWJsaWMvaW1hZ2VzL2NvbW1vbi9kZWZhdWx0LnBuZyIsIng3MjAiOiIvcHVibGljL2ltYWdlcy9jb21tb24vZGVmYXVsdC5wbmcifSwiX2lkIjoiNWNiMjhkZjczOWE0NWVjYzgzNGM1Y2VjIiwicm9sZXMiOlsidXNlciJdLCJpYXQiOjE1NTUyMDU2MjN9.xXno3fV5x5H4546kqyuyGO4S5bjATilSkVCxHGdYNSw\\"
      }
    }
  }
}"
`;
